ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -373,6 +388,11 @@ private[platform] object MutableCacheBackedContractStore {
         "Cannot lookup the maximum ledger time for an empty set of contract identifiers"
       )
 
+  final case class ContractReadThroughNotFound(contractId: ContractId) extends NoStackTrace {
+    override def getMessage: String =
+      s"Contract not found for contract id $contractId. This is likely due to a race condition."
```

Comment:
```suggestion
      s"Contract not found for contract id $contractId. Hint: this could be due racing with a concurrent archival."
```

Reply:
Weaken the text and clarify the kind of race condition to guide the user. In particular, avoid the bad connotation that the server itself has a race condition that leads to a bug.